tests/foreman/cli/test_subnet.py

```diff
@@ -581,3 +588,97 @@ def test_positive_delete_subnet_parameter_overrided_host_impact():
 
     :BZ: 1426612
     """
+
+
+def test_positive_subnet_CRUD_parameters(module_target_sat, module_org, module_location):
+    """Create subnet, set parameters and delete subnet
+
+    :id: 17e039d3-2f7a-4d36-a952-47ca67514ca1
+
+    :steps:
+
+        1. Create valid subnet
+        2. Add valid parameter to subnet
+        3. Delete the above parameter from subnet
+        4. Delete subnet
+
+    :expectedresults: Subnet is created, set with parameters and deleted
+
+    """
+    name = gen_choice(list(valid_data_list().values()))
+    pool = sorted(valid_addr_pools()[0])
+    mask = '255.255.255.0'
+    # generate pool range from network address
+    network = gen_ipaddr()
+    from_ip = re.sub(r'\d+$', str(pool[0]), network)
+    to_ip = re.sub(r'\d+$', str(pool[1]), network)
+    domain = module_target_sat.cli_factory.make_domain(
+        {'location-ids': module_location.id, 'organization-ids': module_org.id}
+    )
+    gateway = gen_ipaddr(ip3=True)
+    ipam_type = SUBNET_IPAM_TYPES['dhcp']
+    # Create subnet
+    subnet = module_target_sat.cli_factory.make_subnet(
+        {
+            'name': name,
+            'from': from_ip,
+            'mask': mask,
+            'network': network,
+            'to': to_ip,
+            'domain-ids': [domain['id']],
+            'gateway': gateway,
+            'ipam': ipam_type,
+            'organization-id': module_org.id,
+            'location-id': module_location.id,
+        }
+    )
+    assert subnet['name'] == name
+    assert subnet['start-of-ip-range'] == from_ip
+    assert subnet['end-of-ip-range'] == to_ip
+    assert domain['name'] in subnet['domains']
+    assert gateway in subnet['gateway-addr']
+    assert ipam_type in subnet['ipam']
+
+    name = gen_string('alphanumeric', 10)
+    value = gen_string('alphanumeric', 10)
+    # Set parameters to subnet
+    module_target_sat.cli.Subnet.set_parameter(
+        {'subnet-id': subnet['id'], 'name': name, 'value': value}
+    )
+    # Fetch subnet info
+    subnet = module_target_sat.cli.Subnet.info({'id': subnet['id']}, output_format='json')
+    assert name in subnet['parameters'][0]['name']  # Now preserves original case
+    assert value in subnet['parameters'][0]['value']
+    assert "string" in subnet['parameters'][0]['parameter_type']
+
+    # Update parameter
+    updated_value = gen_string('alphanumeric', 10)
+    module_target_sat.cli.Subnet.set_parameter(
+        {'subnet-id': subnet['id'], 'name': name, 'value': updated_value}
+    )
+    # Fetch subnet info
+    subnet = module_target_sat.cli.Subnet.info({'id': subnet['id']}, output_format='json')
+    assert name in subnet['parameters'][0]['name']  # Now preserves original case
+    assert updated_value in subnet['parameters'][0]['value']
```

Comment:
```suggestion
    subnet = module_target_sat.cli.Subnet.info({'id': subnet['id']}, output_format='json')
    assert updated_value in subnet['parameters'][0]['value']
```
There's no need to mention '`Fetch subnet info`' every time. Also, there's no need to assert the name again, as it's already being handled. Since we're updating the value here, asserting the value alone should be sufficient.